core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authorization;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Default Access control list for a directory.
+ */
+public class DefaultAccessControlList extends AccessControlList {
+  private AccessControlList mAccessAcl;
+
+  /**
+   * Constructor to build a default ACL based on an access ACL.
+   * @param acl the access acl
+   */
+  public  DefaultAccessControlList(AccessControlList acl) {
+    super();
+    mAccessAcl = acl;
+    mOwningUser = acl.mOwningUser;
+    mOwningGroup = acl.mOwningGroup;
+    mUserActions.put(OWNING_USER_KEY, new AclActions(acl.mUserActions.get(OWNING_USER_KEY)));
+    mGroupActions.put(OWNING_GROUP_KEY, new AclActions(acl.mGroupActions.get(OWNING_GROUP_KEY)));
+    mOtherActions = new AclActions(acl.mOtherActions);
+  }
+
+  private void fillDefault() {
+    if (mUserActions.get(OWNING_USER_KEY) == null) {
```

Comment:
Why does it not replace blindly?

Reply:
i thought not replacing it blindly is the safer option?